src/components/MultiFileInput/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { useCallback } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import InputContainer, { Props as InputContainerProps } from '#components/InputContainer';
+import { NameType } from '#components/types';
+
+import { CloseLineIcon } from '@ifrc-go/icons';
+import IconButton from '#components/IconButton';
+import RawFileInput, { RawFileInputProps } from '../RawFileInput';
+import styles from './styles.module.css';
+
+type InheritedProps<N extends NameType> = Omit<InputContainerProps, 'input'> & Omit<RawFileInputProps<N>, 'multiple' | 'onChange' | 'children'>;
+
+export type MultiFileInputProps<N extends NameType> = InheritedProps<N> & {
+    valueComponent?: React.FC<{ value: File[] | File | null }>;
+    clearable?: boolean;
+    clearButtonProps?: React.ComponentPropsWithoutRef<'button'>;
+    inputClassName?: string;
+    fileInputProps?: React.ComponentPropsWithoutRef<'input'>;
+    placeholder?: React.ReactNode;
+    value: File[] | null;
+    onChange: (files: File[] | null, name: N) => void;
+}
+
+interface DefaultValueProps {
+    value: File[] | null
+}
+
+function DefaultValue(props: DefaultValueProps) {
+    const { value } = props;
+    return (
+        <div>
+            {value?.map((file) => file.name).join(', ')}
+        </div>
+    );
+}
+
+function MultiFileInput<N extends NameType>(props: MultiFileInputProps<N>) {
+    const {
+        accept,
+        actions: actionsFromProps,
+        capture,
+        className,
+        clearButtonProps,
+        clearable,
+        disabled,
+        error,
+        errorOnTooltip,
+        fileInputProps,
+        form,
+        hint,
+        icons,
+        inputClassName,
+        inputProps,
+        inputSectionClassName,
+        label,
+        name,
+        onChange,
+        placeholder,
+        readOnly,
+        required,
+        value,
+        valueComponent,
+        variant,
+        withAsterisk,
+        ...others
+    } = props;
+
+    const ValueComponent = valueComponent ?? DefaultValue;
+    const hasValue = Array.isArray(value) ? value.length !== 0 : value !== null;
+
+    const handleClear = useCallback(() => {
+        onChange(null, name);
```

Comment:
We should not use null as a return value from input. Let's instead use undefined.